From a7d1ea3e37ad1dbbe9d080eebe10b5e5f5f42c36 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 12 Jan 2025 15:43:08 +0800
Subject: [PATCH 177/301] AOSCOS: kvm: disable enable_virt_at_load by default

Disable enable_virt_at_load, a >= 6.12 on-by-default parameter introduced
with commit b4886fab6fb6 ("KVM: Add a module param to allow enabling
virtualization when KVM is loaded"). By enabling this paramter, VirtualBox
(as of 7.1.4) will fail to launch any virtualised systems with the
following error:

  VirtualBox can't operate in VMX root mode. Please disable the KVM kernel
  extension, recompile your kernel and reboot (VERR_VMX_IN_VMX_ROOT_MODE).

The original commit indicated that this was introduced to reduce latency
during KVM initialisation and that it "could be problematic for use case
that need to spin up VMs quickly." An optimisation that is more than
likely unnecessary for our (predominantly desktop-oriented) crowd.

Bottom line - make VirtualBox work again (until they come up with a fix).

Fixes: b4886fab6fb6 ("KVM: Add a module param to allow enabling virtualization when KVM is loaded")
Reported-by: xtex <xtexchooser@duck.com>
Link: https://github.com/AOSC-Dev/aosc-os-abbs/issues/8897
Link: https://lore.kernel.org/kvm/ZwQjUSOle6sWARsr@google.com/T/
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Link: https://t.me/c/1109254909/715020
Acked-by: xtex <xtexchooser@duck.com>
Reviewed-by: xtex <xtexchooser@duck.com>
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 virt/kvm/kvm_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/virt/kvm/kvm_main.c b/virt/kvm/kvm_main.c
index ba0327e2d0d3..c6a7f719b570 100644
--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -5466,7 +5466,7 @@ static struct miscdevice kvm_dev = {
 };
 
 #ifdef CONFIG_KVM_GENERIC_HARDWARE_ENABLING
-static bool enable_virt_at_load = true;
+static bool enable_virt_at_load = false;
 module_param(enable_virt_at_load, bool, 0444);
 
 __visible bool kvm_rebooting;
-- 
2.48.1

